File: Compose-Bookshelf-App/app/src/main/java/com/example/bookshelfapp/model/GoogleBookRetrieveApiModel.kt
package com.example.bookshelfapp.model

import kotlinx.serialization.Serializable

@Serializable
data class GoogleBookRetrieveApiModel(
    val volumeInfo: RetrieveVolumeInfoModel
)

@Serializable
data class RetrieveVolumeInfoModel(
    val imageLinks: ThumbnailModel? = null
)

@Serializable
data class ThumbnailModel(
    val thumbnail: String
)

File: SwiftUI-Bookshelf-App/Bookshelf app/Model/GoogleBookRetrieveApiModel.swift
//
//  GoogleBookRetrieveApiModel.swift
//  Bookshelf app
//
//  Created by Jaehwa Noh on 1/27/24.
//

struct GoogleBookRetrieveApiModel: Decodable {
    let volumeInfo: RetrieveVolumeInfoModel
}

struct RetrieveVolumeInfoModel: Decodable {
    let imageLinks: ThumbnailModel?
}

struct ThumbnailModel: Decodable {
    let thumbnail: String
}
